storage/storage_client.go

```diff
@@ -101,7 +102,7 @@ func (c *StorageClient) getUnqualifiedName(name string) string {
 	}
 
 	nameParts := strings.Split(name, "/")
-	return strings.Join(nameParts[2:], "/")
+	return strings.Join(nameParts[len(nameParts)-1:], "/")
```

Comment:
Ah, meant we should check that the length of `nameParts` is actually correct, before indexing. Shouldn't ever be incorrect, but better to protect against it now, than deal with a potential panic in the future. 

```
if len(nameParts) < 2 {
  // raise error
}
```